stumpy/motifs.py

```diff
@@ -0,0 +1,412 @@
+
+    max_motifs : int, default 1
+        Maxmimum number of motifs to return.
+
+    normalize : bool, default True
+        When set to `True`, this z-normalizes subsequences prior to computing distances.
+        Otherwise, this function gets re-routed to its complementary non-normalized
+        equivalent set in the `@core.non_normalized` function decorator.
+
+    Return
+    ------
+    top_motif_indices : ndarray
+        Array of top motif indices. List of the indices of all matches of the top
+        k motifs, sorted by distance from the motif representative. The motif
+        representative starts at the first index and is the subsequence with the lowest
+        matrix profile value that is part of a motif.
+
+        There are at most `k` rows, each of which represents
+        one motif. All found matches within a distance of `atol + MP * rtol` (where
+        MP is the matrix profile value of the motif representative) are returned. The
+        array has either `max_neighbors` columns if specified, or as many columns as
+        number of matches for the motif with most found matches. Everything else
+        is filled up with -1.
+
+    top_motif_values : ndarray
+        For every match its distance to the motif representative.
+    """
+    if max_motifs < 1:  # pragma: no cover
+        logger.warn(
+            "The maximum number of motifs, `max_motifs`, "
+            "must be greater than or equal to 1"
+        )
+        logger.warn("`max_motifs` has been set to `1`")
+        max_motifs = 1
+
+    if T.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {T.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    if P.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {P.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    m = T.shape[0] - P.shape[0] + 1
+    if excl_zone is None:  # pragma: no cover
+        excl_zone = int(np.ceil(m / 4))
+    if max_matches is None:  # pragma: no cover
+        max_matches = np.inf
+    if cutoff is None:  # pragma: no cover
+        cutoff = max(np.mean(P) - 2 * np.std(P), np.min(P))
+
+    # core.preprocess performs a copy, so modifiying this T will not
+    # change the function input time series
+    T, M_T, Σ_T = core.preprocess(T[np.newaxis, :], m)
+
+    P = P.copy().astype(float)
+    P[P > cutoff] = np.inf
+    P = P[np.newaxis, :]
+
+    result = _motifs(
+        T=T,
+        P=P,
+        M_T=M_T,
+        Σ_T=Σ_T,
+        excl_zone=excl_zone,
+        min_neighbors=min_neighbors,
+        max_distance=max_distance,
+        max_matches=max_matches,
+        max_motifs=max_motifs,
+        normalize=normalize,
+    )
+    result = (
+        _jagged_list_to_array(result[0], fill_value=-1, dtype="int64"),
```

Comment:
To be consistent with matrix profiles, I feel like we should make the array the distances and the second array the indices.